corokafka/corokafka_header_pack.cpp

```diff
@@ -55,38 +56,46 @@ HeaderPack& HeaderPack::pop_back() {
     return *this;
 }
 
-void HeaderPack::erase(const std::string& name) {
-    _headers.erase(std::remove_if(_headers.begin(), _headers.end(), [&name](const HeaderNode& entry)->bool {
-        return StringEqualCompare()(entry.first, name);
+void HeaderPack::erase(const std::string& name, size_t relativePosition) {
+    size_t pos = 1;
+    _headers.erase(std::remove_if(_headers.begin(), _headers.end(), [&](const HeaderNode& header)->bool {
+        if (StringEqualCompare()(header.first, name)) {
+            return (relativePosition == 0) || (pos++ == relativePosition);
```

Comment:
== should be >= if I understand correctly what is being done here.

Reply:
We want to stop when we reach the Nth header with this name. The logic is ok. If 0 is passed, then it's the first occurrence.